src/coverage_reporter/parsers/clover_parser.cr

```diff
@@ -0,0 +1,94 @@
+require "./base_parser"
+require "xml"
+
+module CoverageReporter
+  class CloverParser < BaseParser
+    record Info,
+      coverage : Hash(Line, Hits?),
+      branches : Hash(Line, Array(Hits))
+
+    # NOTE: Provide the base path for the sources. You can check "filename" in
+    #       coverage report and see what part is missing to get a valid source path.
+    def initialize(@base_path : String?)
+    end
+
+    def globs : Array(String)
+      [
+        "**/*/clover.xml",
+        "clover.xml",
+      ]
+    end
+
+    def matches?(filename) : Bool
+      return true if /clover.xml/.matches?(filename)
```

Comment:
We shouldn't check by name here because people can rename the Jacoco file to clover.xml and call the coverage-reporter as `coveralls report clover.xml`. In this case we should check its content instead of name. Name is only important when selecting files via globs (calling as `coveralls report`).